browser/web_discovery/web_discovery_utils.cc

```diff
@@ -0,0 +1,70 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/web_discovery/web_discovery_utils.h"
+
+#include "brave/common/pref_names.h"
+#include "brave/components/search_engines/brave_prepopulated_engines.h"
+#include "components/prefs/pref_service.h"
+#include "components/search_engines/template_url.h"
+#include "components/search_engines/template_url_service.h"
+
+bool IsBraveSearchDefault(TemplateURLService* template_service) {
+  DCHECK(template_service);
+  if (!template_service)
+    return false;
+
+  auto* template_url = template_service->GetDefaultSearchProvider();
+  if (!template_url)
+    return false;
+  return template_url->prepopulate_id() ==
+         TemplateURLPrepopulateData::PREPOPULATED_ENGINE_ID_BRAVE;
+}
+
+bool ShouldCreateWebDiscoveryTabHelper(PrefService* prefs) {
+  if (prefs->GetBoolean(kDontAskEnableWebDiscovery))
+    return false;
+
+  if (prefs->GetInteger(kBraveSearchVisitCount) == 20)
+    return false;
+
+  return true;
+}
+
+bool ShouldShowWebDiscoveryDialog(PrefService* prefs,
+                                  TemplateURLService* service) {
+  DCHECK(prefs && service);
+  if (!prefs || !service)
+    return false;
+
+  if (!IsBraveSearchDefault(service))
+    return false;
+
+  if (prefs->GetBoolean(kWebDiscoveryEnabled))
+    return false;
+
+  const int visit_count = prefs->GetInteger(kBraveSearchVisitCount);
+  return (visit_count == 3 || visit_count == 10 || visit_count == 20);
+}
+
+bool IncreaseBraveSearchVisitCount(PrefService* prefs,
+                                   TemplateURLService* service) {
+  DCHECK(prefs && service);
+  if (!prefs || !service)
+    return false;
+
+  if (!IsBraveSearchDefault(service))
+    return false;
+
+  if (prefs->GetBoolean(kWebDiscoveryEnabled))
+    return false;
+
+  int visit_count = prefs->GetInteger(kBraveSearchVisitCount);
+  // Don't need to increase anymore. We don't show again after 20th visit.
+  if (visit_count >= 20)
```

Comment:
It might also be simpler to just not bother with this condition here... Every other place in the code can look for `>` 20. It's also weird because you also pop the dialog when visit count is 20.